### PR TITLE
DQ's page small PRs

### PR DESCRIPTION
**Pull request by sudheer-ajax**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-5409

### Change description

DQ's page small PRs

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/directions-questionnaire/routes/hearing-exceptional-circumstances.ts [hunk 1]
```diff
@@ -34,21 +41,42 @@ export default express.Router()
...
         const user: User = res.locals.user
+        if (party === MadeBy.CLAIMANT && res.locals.claim.response.directionsQuestionnaire) {
+          const defendantDirectionsQuestionnaire: DirectionsQuestionnaire = res.locals.claim.response.directionsQuestionnaire
 
+          if (form.model.exceptionalCircumstances.option === YesNoOption.YES.option) {
+            draft.document.hearingLocation.courtName = defendantDirectionsQuestionnaire.hearingLocation.courtName
+            form.model.reason = undefined
+          }
+        } else {
+          if (form.model.exceptionalCircumstances.option === YesNoOption.NO.option) {
```
Comment (robertparkinson) [inline src/main/features/directions-questionnaire/routes/hearing-exceptional-circumstances.ts]: can you please make this an else if? no need to have the if inside the else if there's no other options

File: src/main/features/directions-questionnaire/routes/hearing-location.ts [hunk 2]
```diff
@@ -97,17 +114,38 @@ export default express.Router()
...
+              draft.document.hearingLocationSlug = court.slug
               form.model = new HearingLocation(court.name, form.model.alternativePostcode, alternativeCourtDetails.facilities)
             }
+            await new DraftService().save(draft, user.bearerToken)
             renderPage(res, form, court === undefined)
+          } else if (form.model.courtAccepted === YesNoOption.NO && form.model.alternativeOption === AlternativeCourtOption.BY_NAME) {
+            draft.document.hearingLocation = form.model
+            draft.document.hearingLocation.alternativeOption = form.model.alternativeOption
+            draft.document.hearingLocation.courtName = form.model.alternativeCourtName
+            draft.document.hearingLocationSlug = ''
```
Comment (robertparkinson) [inline src/main/features/directions-questionnaire/routes/hearing-location.ts]: can all these empty strings be replaced with undefined please?

File: src/main/features/directions-questionnaire/routes/hearing-location.ts [hunk 3]
```diff
@@ -97,17 +114,38 @@ export default express.Router()
...
+            await new DraftService().save(draft, user.bearerToken)
+            res.redirect(Paths.expertPage.evaluateUri({ externalId: res.locals.claim.externalId }))
           } else {
-
-            if (form.model.courtAccepted === undefined) {
-              draft.document.hearingLocation.alternativeCourtName = form.model.alternativeCourtName
-            } else {
-              draft.document.hearingLocation.courtName = form.model.courtName
+            if (form.model.alternativeOption !== undefined && form.model.alternativeOption === AlternativeCourtOption.BY_NAME) {
+              draft.document.hearingLocation.alternativeCourtName = ''
```
Comment (robertparkinson) [inline src/main/features/directions-questionnaire/routes/hearing-location.ts]: can all these empty strings be replaced with undefined please?

File: src/main/features/directions-questionnaire/routes/hearing-location.ts [hunk 4]
```diff
@@ -97,17 +114,38 @@ export default express.Router()
...
+              draft.document.hearingLocation.alternativeOption = undefined
+            } else if (!draft.document.hearingLocationSlug.length) {
+              const postcode: string = getDefaultPostcode(res)
+              const court: Court = await getNearestCourt(postcode)
+              if (court !== undefined) {
+                draft.document.hearingLocationSlug = court.slug
+              } else {
+                draft.document.hearingLocationSlug = ''
```
Comment (robertparkinson) [inline src/main/features/directions-questionnaire/routes/hearing-location.ts]: can all these empty strings be replaced with undefined please?

File: src/main/features/directions-questionnaire/views/hearing-location.njk [hunk 5]
```diff
@@ -52,20 +52,20 @@
...
+                  checked: form.valueFor('courtAccepted').option === YesNoOption.NO.option
                 }
               ]
             ) }}
             <div class="js-hidden" id="courtAccepted-no">
               <h1 class="heading-medium">{{ t('Choose another court') }}</h1>
               {{ radioOption(form = form, label = 'Enter a court name', name = 'alternativeOption', value = 'name', checked = form.valueFor('alternativeOption') === 'name', bold = true) }}
               <div class="form-group panel panel-border-narrow js-hidden " id="alternativeOption-name" aria-hidden="false">
-                {{ textInput(label = 'Court name', form = form, name = 'alternativeCourtName', inputClass = 'full-width') }}
+                {{ textInput(label = 'Court name', form = form, name = 'alternativeCourtName', inputClass = 'full-width') }}Do you want to choose
```
Comment (robertparkinson) [inline src/main/features/directions-questionnaire/views/hearing-location.njk]: this looks like a copy/paste gone bad. Please remove

File: src/main/features/directions-questionnaire/routes/hearing-location.ts (hunk 4 above)
Comment (sudheer-ajax) [reply src/main/features/directions-questionnaire/routes/hearing-location.ts]: hearingLocationSlug is defined as empty string by default directionsQuestionnaireDraft.

File: src/main/features/directions-questionnaire/routes/hearing-location.ts (hunk 3 above)
Comment (sudheer-ajax) [reply src/main/features/directions-questionnaire/routes/hearing-location.ts]: As we discussed before that courtNames are defined as Mandatory .... I have added todo for this to remove or undefined once backend is fixed

File: src/main/features/directions-questionnaire/routes/hearing-location.ts (hunk 2 above)
Comment (sudheer-ajax) [reply src/main/features/directions-questionnaire/routes/hearing-location.ts]: hearingLocationSlug is defined as empty string by default directionsQuestionnaireDraft.

File: src/main/features/directions-questionnaire/routes/hearing-exceptional-circumstances.ts (hunk 1 above)
Comment (sudheer-ajax) [reply src/main/features/directions-questionnaire/routes/hearing-exceptional-circumstances.ts]: done

File: src/main/features/directions-questionnaire/views/hearing-location.njk (hunk 5 above)
Comment (sudheer-ajax) [reply src/main/features/directions-questionnaire/views/hearing-location.njk]: done

**Pull request closed without merging**